simulationworkflowschema/general.py

```diff
@@ -231,24 +285,214 @@ def normalize(self, archive, logger):
+            except Exception:
+                continue
+
+    def get_gw_workflow_results(
+        self, initial_task: TaskReference, final_task: TaskReference
+    ) -> None:
+        """
+        Gets the GW workflow results section by resolving the DFTOutputs from the initial
+        task and the GWOutputs from the final task.
+        """
+        dft_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(dft_outputs, initial_task)
+        gw_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(gw_outputs, final_task)
+        self.results.dft_outputs = dft_outputs
+        self.results.gw_outputs = gw_outputs
+
+    def get_dmft_workflow_results(
+        self, initial_task: TaskReference, final_task: TaskReference
+    ) -> None:
+        """
+        Gets the DMFT workflow results section by resolving the TBOutputs from the initial
+        task and the DMFTOutputs from the final task.
+        """
+        tb_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(tb_outputs, initial_task)
+        dmft_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(dmft_outputs, final_task)
+        self.results.tb_outputs = tb_outputs
+        self.results.dmft_outputs = dmft_outputs
+
+    def get_maxent_workflow_results(
+        self, initial_task: TaskReference, final_task: TaskReference
+    ) -> None:
+        """
+        Gets the MaxEnt workflow results section by resolving the DMFTOutputs from the
+        initial task and the MaxEntOutputs from the final task.
+        """
+        dmft_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(dmft_outputs, initial_task)
+        maxent_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(maxent_outputs, final_task)
+        self.results.dmft_outputs = dmft_outputs
+        self.results.maxent_outputs = maxent_outputs
+
+    def get_tb_workflow_results(
+        self, initial_task: TaskReference, final_task: TaskReference
+    ) -> None:
+        """
+        Gets the TB workflow results section by resolving the FirstPrinciplesOutputs from the
+        initial task and the TBOutputs from the final task.
+        """
+        first_principles_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(first_principles_outputs, initial_task)
+        tb_outputs = ElectronicStructureOutputs()
+        self._resolve_outputs_section(tb_outputs, final_task)
+        self.results.first_principles_outputs = first_principles_outputs
+        self.results.tb_outputs = tb_outputs
+
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+        if len(self.tasks) != 2:
+            logger.error("Expected two tasks.")
+            return
+
+        initial_task = self.tasks[0]
+        final_task = self.tasks[1]
+
+        workflow_name = self.m_def.name.replace("Plus", "+")
```

Comment:
i really do not like forcing this convention. It would rather have a mapping of names.

Reply:
(they are equally shit, imho 😆 )

Reply:
Maybe there are methods containing 'Plus'. Generally, I do not like anything that operates on strings. How about extracting name from the workflow tasks.

Reply:
Well, I am talking about naming classes too.

Reply:
But the reason you put the Plus is to derive the workflow name. I would like to prevent a Plus runaway catastrophe.

Reply:
Ok, I was talking with Joe. There are no good options:

- We can name everything with special names (like `GWWorkflow` or `DMFTWorkflow`).
- We can name everything with flattened acronyms (`DFTGW`, `DFTTBDMFT`).
- We can try to separate the acronyms (`DFTPlusGW`, `DFTXGW`, or even use underscore `DFT_GW`?)

In any case, imagine a situation where we have a `DFT+TB+DMFT` and a `DMFT+MaxEnt` workflow, and then we are able to connect both types of workflows. In all 3 options I said, how would you call all these 3 workflows? Or should we deprecate two of them and generate a 3rd more general one (with the problems that that carries)? For me `Plus` makes more sense after thinking, because it can be extended (although with horrible naming...) and it is more natural (experts know what a "DFT+TB+DMFT" calculation is).

But again, these options are horrible anyways.

😢

Reply:
I refer this then to my comment about not creating classes for these kinds of workflows but only connect them thru a generic workflow entry. Ok, we can do Plus, but to derive workflow name please simply concatenate the tasks names.

Reply:
Ok

Reply:
Can you check now? I added a condition on checking the `tasks.name` and even comparing with `workflow.m_def.name` for consistency.